### ISSUE TITLE
Redirect Url on logout 

### ISSUE THREAD

**Issue by zebamba** [IdentityServer/IdentityServer3#1968]
This is the same issue as #1827 still happening. 

I upgraded Thinktecture.IdentityServer3 v1.6.2 to IdentityServer3 v2.0.0.

https://github.com/IdentityServer/IdentityServer3/blob/master/source/Core/Endpoints/Connect/EndSessionController.cs

My guess is that after logout it calls 

<pre>
        [Route(Constants.RoutePaths.Oidc.EndSessionCallback, Name = Constants.RouteNames.Oidc.EndSessionCallback)]
        [HttpGet]
        public IHttpActionResult LogoutCallback()
        {
            Logger.Info("End session callback requested");
            
            return ResponseMessage(new HttpResponseMessage(HttpStatusCode.OK)
            {
                Content = new StringContent(String.Empty, Encoding.UTF8, "text/html")
            });
        }
</pre>

What return an empty String. I hope I am wrong on this. 

Anyways on the client View

    public Task<Stream> LoggedOut(LoggedOutViewModel model, SignOutMessage message)
    {
        return Render(model, "loggedOut");
    }

The LoggedOutViewModel the property IFrameUrls has this value [0] = "https://localhost:44306/connect/endsessioncallback"

protect virtual Task Render(CommonViewModel model ... )

message is null
ClientName is null
RedirectUrl  is null

As the RedirectUrl is null passing to app.js the redirect fails on code below as
model.redirectUrl is undefined.

<pre>
if (model.autoRedirect && model.redirectUrl) 
{
        if (model.autoRedirectDelay < 0) {
            model.autoRedirectDelay = 0;
        }
        window.setTimeout(function () {
            window.location = model.redirectUrl;
        }, model.autoRedirectDelay * 1000);
}
</pre>

**Comment by brockallen** [comment]
For client and redirect to be filled in the client app must send the the id token hint and the post logout redirect uri that matches one in the client config. Is the client app doing that? If so, then check the logs to see if they're not validating properly.

**Comment by zebamba** [comment]
It is all passing correct and values match database.

How do I turn log on and check it?

Thank you

**Comment by brockallen** [comment]
Check the docs to learn how to enable logging.

**Comment by kalanbates** [comment]
I am in a very similar spot as the OP.  My current implementation is based on `IdentityServer3 Version 2.0.0-build00174` with the `Samples.Clients.Javascript Implicit Client` as my test client.  I've made a small modification to some of the client script there to autoconfigure my Logout function, which is shared below.

```
app.factory("endSessionEndpoint", function ($http) {
            var url = "https://localhost:44333/core/.well-known/openid-configuration";
            console.log("[$scope.endSessionEndpoint is accessible]");
            return $http.get(url).then(function (openid_configuration) {
                console.log("[openid-configuration]: " + JSON.stringify(openid_configuration.data));
                return openid_configuration.data.end_session_endpoint;
            });
        });
```
I'm storing the `response.id_token` that comes back from the authorization endpoint as `$scope.response.id_token_hint`

Following Dominick's recommendations on handling <a href="http://leastprivilege.com/2014/10/14/identityserver-v3-and-post-logout-redirect/">Post Logout Redirects</a>, I am sending down both my `id_token_hint` and my `post_logout_redirect_uri` but I end up being internally redirected to `/core/logout`

```
$scope.logout = function (){
                 endSessionEndpoint.then(function (endpoint){
                     console.log("[Current location] "+ $window.location);
                     console.log("[EndSession endpoint] " + endpoint);
                     var uri = "{0}?id_token_hint={1}&post_logout_redirect_uri={2}"
                                    .format(endpoint,
                                            $scope.response.id_token_hint, 
                                            $window.location.origin);
                     $window.location = uri;
                 });
            }
```

...it's pretty ugly, but just for completeness' sake, here is the URL that I'm hitting the endpoint with:
(yes, I realize this is a horribly large id_token...it has a lot of very smelly legacy claims included in it. `post_logout_redirect_uri` is at the end)
```
https://localhost:44333/core/connect/endsession?id_token_hint=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImEzck1VZ01Gdjl0UGNsTGE2eUYzekFrZnF1RSIsImtpZCI6ImEzck1VZ01Gdjl0UGNsTGE2eUYzekFrZnF1RSJ9.eyJub25jZSI6IjkxMTgzMTI3MzkzMTE4MTgiLCJpYXQiOjE0NDM4MDgxNDAsImF0X2hhc2giOiJReUNVQzRSRWRWdmxuN09yWWRabjhBIiwic3ViIjoia2V2aW5xdWF5IiwiYW1yIjoicGFzc3dvcmQiLCJhdXRoX3RpbWUiOjE0NDM4MDgxNDAsImlkcCI6Imlkc3J2IiwibmFtZSI6ImtldmlucXVheSIsInJlYWxtIjoibGVnYWN5Iiwic3luY2hjb2RlIjoiZTIzYjcyY2QtZmQzNy00N2RiLWJiNzAtOGZkNmEwMGQ5OThjIiwiY29ycmVsYXRvciI6Ijk3NWM5NjM2LTQyZTYtNDk5Zi1iYzQwLTlkNTA5YmIxYmQzNiIsInJvbGUiOiJ7IFJvbGUgPSBDdXN0b21lciwgQ29kZSA9IFJPTEVfRUNPX0NVU1RPTUVSIH0iLCJwZXJtaXNzaW9ucyI6IltcImxua0FjY291bnRNYW5hZ2VtZW50SG9tZVwiLFwibG5rUmVuZXdBZ3JlZW1lbnRcIixcImxua0FkZFNlcnZpY2VMb2NhdGlvblwiLFwibG5rVmlld0FsbFN0YXRlbWVudHNcIixcImxua1ZpZXdDdXJyZW50U3RhdGVtZW50XCIsXCJsbmtNYWtlYVBheW1lbnRcIixcImxua0NvbXBsZXRlRW5yb2xsbWVudFwiLFwibG5rTW9kaWZ5QXV0b1BheVwiLFwibG5rVXNhZ2VTdGF0aXN0aWNzXCIsXCJsbmtWaWV3U3VtbWFyeVwiLFwibG5rTW9kaWZ5TG9naW5JbmZvcm1hdGlvblwiLFwibG5rQ3VzdG9taXplQWNjb3VudE5hbWVcIixcImxua0N1c3RvbWl6ZU1ldGVyTmFtZXNcIixcImxua1ZpZXdZb3VyQWdyZWVtZW50XCIsXCJsbmtTdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwibG5rVHJhbnNhY3Rpb25IaXN0b3J5XCIsXCJsbmtQSk1NSVNPSW5kZXhQcmljaW5nXCIsXCJsbmtSZWZlcmFGcmllbmRcIixcInR4dE90aGVyQW1vdW50X01ha2VhUGF5bWVudFwiLFwic2xCYW5rQWNjb3VudFR5cGVfTWFrZVBheW1lbnRcIixcInR4dEFjY291bnRGaXJzdE5hbWVfTWFrZVBheW1lbnRcIixcInR4dEFjY291bnRMYXN0TmFtZV9NYWtlUGF5bWVudFwiLFwidHh0QWNjb3VudEhvbGRlck5hbWVfTWFrZVBheW1lbnRcIixcInR4dFJvdXRpbmdOdW1iZXJfTWFrZVBheW1lbnRcIixcInR4dEFjY291bnROdW1iZXJfTWFrZVBheW1lbnRcIixcInR4dEFjY291bnRBZGRyZXNzMV9NYWtlUGF5bWVudFwiLFwidHh0QWNjb3VudEFkZHJlc3MyX01ha2VQYXltZW50XCIsXCJ0eHRDaXR5X01ha2VQYXltZW50XCIsXCJzbFN0YXRlX01ha2VQYXltZW50XCIsXCJ0eHRaSVBDb2RlX01ha2VQYXltZW50XCIsXCJsbmtPcHRpb25zX01ha2VQYXltZW50XCIsXCJidG5Db250aW51ZV9NYWtlUGF5bWVudFwiLFwibG5rQ2FuY2VsX01ha2VQYXltZW50XCIsXCJidG5TdWJtaXRQYXltZW50X01ha2VQYXltZW50XCIsXCJsbmtFZGl0SW5mb3JtYXRpb25fTWFrZVBheW1lbnRcIixcImxua1ByaW50UmVjZWlwdF9NYWtlUGF5bWVudFwiLFwic2xCYW5rQWNjb3VudFR5cGVfQXV0b3BheVwiLFwidHh0QWNjb3VudEZpcnN0TmFtZV9BdXRvcGF5XCIsXCJ0eHRBY2NvdW50TGFzdE5hbWVfQXV0b3BheVwiLFwidHh0QWNjb3VudEhvbGRlck5hbWVfQXV0b3BheVwiLFwidHh0Um91dGluZ051bWJlcl9BdXRvcGF5XCIsXCJ0eHRBY2NvdW50TnVtYmVyX0F1dG9wYXlcIixcInR4dEFjY291bnRBZGRyZXNzMV9BdXRvcGF5XCIsXCJ0eHRBY2NvdW50QWRkcmVzczJfQXV0b3BheVwiLFwidHh0Q2l0eV9BdXRvcGF5XCIsXCJzbFN0YXRlX0F1dG9wYXlcIixcInR4dFpJUENvZGVfQXV0b3BheVwiLFwibG5rT3B0aW9uc19BdXRvcGF5XCIsXCJsbmtNb2RpZnlBdXRvcGF5SW5mb3JtYXRpb25fQXV0b3BheVwiLFwibG5rQ2FuY2VsQXV0b3BheUVucm9sbG1lbnRfQXV0b3BheVwiLFwiYnRuQ29udGludWVfQXV0b3BheVwiLFwibG5rQ2FuY2VsX0F1dG9wYXlcIixcImJ0bkFjdGl2YXRlX0F1dG9wYXlcIixcImJ0bkVkaXRJbmZvcm1hdGlvbl9BdXRvcGF5XCIsXCJzbG1ldGVyU2VsZWN0ZWRfVXNhZ2VTdGF0aXN0aWNzXCIsXCJ0eHRVc2VyTmFtZV9Nb2RpZnlMb2dpbkluZm9ybWF0aW9uXCIsXCJ0eHRQYXNzd29yZF9Nb2RpZnlMb2dpbkluZm9ybWF0aW9uXCIsXCJ0eHRDb25maXJtUGFzc3dvcmRfTW9kaWZ5TG9naW5JbmZvcm1hdGlvblwiLFwidHh0RW1haWxfTW9kaWZ5TG9naW5JbmZvcm1hdGlvblwiLFwic2xRdWVzdGlvbnMwMV9Nb2RpZnlMb2dpbkluZm9ybWF0aW9uXCIsXCJ0eHRBbnN3ZXIwMV9Nb2RpZnlMb2dpbkluZm9ybWF0aW9uXCIsXCJzbFF1ZXN0aW9uczAyX01vZGlmeUxvZ2luSW5mb3JtYXRpb25cIixcInR4dEFuc3dlcjAyX01vZGlmeUxvZ2luSW5mb3JtYXRpb25cIixcInNsUXVlc3Rpb25zMDFfTW9kaWZ5TG9naW5JbmZvcm1hdGlvbkNvbVwiLFwiYnRuU2F2ZV9Nb2RpZnlMb2dpbkluZm9ybWF0aW9uXCIsXCJ0eHRBY2NvdW50TmFtZV9DdXN0b21pemVBY2NvdW50TmFtZVwiLFwiYnRuQWNjb3VudE5hbWVfQ3VzdG9taXplQWNjb3VudE5hbWVcIixcInNsTWV0ZXJMaXN0X0N1c3RvbWl6ZU1ldGVyTmFtZVwiLFwidHh0TWV0ZXJOYW1lX0N1c3RvbWl6ZU1ldGVyTmFtZVwiLFwiYnRuU2F2ZU1ldGVyTmFtZV9DdXN0b21pemVNZXRlck5hbWVcIixcImxua0JpbGxpbmdQcmVmZXJlbmNlc19TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwidHh0QXJlYUVtYWlsX1N0YXRlbWVudFByZWZlcmVuY2VzXCIsXCJ0eHRGYXhBcmVhQ29kZV9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwidHh0RmF4Q29udGFjdE51bWJlcl9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwidHh0RmF4Q29udGFjdE51bWJlcjJfU3RhdGVtZW50UHJlZmVyZW5jZXNcIixcInR4dEF0dGVudGlvblRvRmF4X1N0YXRlbWVudFByZWZlcmVuY2VzXCIsXCJidG5TYXZlRUJpbGxJbmZvcm1hdGlvbl9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwibG5rQmlsbGluZ0FkZHJlc3NfU3RhdGVtZW50UHJlZmVyZW5jZXNcIixcInR4dEFkZHJlc3NMaW5lMV9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwidHh0QWRkcmVzc0xpbmUyX1N0YXRlbWVudFByZWZlcmVuY2VzXCIsXCJ0eHRDaXR5TmFtZV9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwic2xTdGF0ZV9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwidHh0UG9zdGFsQ29kZV9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwiYnRuU2F2ZVN0YXRlbWVudFByZWZlcmVuY2VzX1N0YXRlbWVudFByZWZlcmVuY2VzXCIsXCJsbmtCdWRnZXRCaWxsaW5nX1N0YXRlbWVudFByZWZlcmVuY2VzXCIsXCJidG5FbnJvbGxCdWRnZXRfU3RhdGVtZW50UHJlZmVyZW5jZXNcIixcImJ0blRlcm1pbmF0ZUJ1ZGdldF9TdGF0ZW1lbnRQcmVmZXJlbmNlc1wiLFwic2xab25lX0luZGV4UHJpY2VcIixcInNsUmVhbFRpbWVfSW5kZXhQcmljZVwiLFwiYnRuU3VibWl0X0luZGV4UHJpY2VcIixcImJ0blJlbmV3QWdyZWVtZW50X1JlbmV3QWdyZWVtZW50XCIsXCJidG5EZWNsaW5lQWdyZWVtZW50X1JlbmV3QWdyZWVtZW50XCIsXCJsbmtQb3N0dG9GYWNlYm9va19SZWZlcmFGcmllbmRcIixcImxua0dlbmVyYXRlRW1haWxfUmVmZXJhRnJpZW5kXCIsXCJidG5TdWJtaXRfUmVmZXJhRnJpZW5kXCIsXCJsbmtTYXZpbmdzXCIsXCJsbmtSZW5ld0FncmVlbWVudEJ5U3BlY2lhbE9mZmVyc1wiLFwiYnRuUmVuZXdBZ3JlZW1lbnRCeVNwZWNpYWxPZmZlcnNcIl0iLCJpc3MiOiJodHRwczovL2xvY2FsaG9zdDo0NDMzMy9jb3JlIiwiYXVkIjoiYWNjb3VudG1hbmFnZW1lbnQiLCJleHAiOjE0NDM4MDg1MDEsIm5iZiI6MTQ0MzgwODE0MX0.T680uo8Q023xtWpamGNInaxOiBAhVAWhrt5FiJ_SLBdsrdYDMjWmM_MGBEIEfAr5-9wuDPaINSaHF7EqBSTYvnK4_I0j0lRf0AULkUseCEfLHdw2aTJjnzCS_Zog_HvCJZc1soNuIiEpUWZfnvynAi5XePjdbOO7ytehAr_j-7TDaSZo4PM-Ra_hjpE3syN0c3CHgdfYzw_Q0O21Cebu32WAwKj-i3O2P9xwGKnTp_fh5BnaoD9bB0sRzBoUcsF4h6ljFwaJE_mvTE6OB0ElV-w0S-bQqgAOGaiSU9WGNmT5UlO_mqoNI1bH0zgaCFvyA_6qKApzkD7tuEjRTw8eNQ&post_logout_redirect_uri=http://localhost:21575
```

and `http://localhost:21575` is a registered callback for my Client.

I get sent to `https://localhost:44333/core/logout`

**Comment by leastprivilege** [comment]
did you enable logging?

**Comment by kalanbates** [comment]
Yes, I have trace logging enabled through Serilog.
I was able to resolve my particular issue.

The root of my problem is that I had configured my redirectURI in the `RedirectUris` section of my Client config instead of configuring the `PostLogoutRedirectUris` section.

@zebamba Are you certain you did not make the same mistake that I made?  Confirm that your `post_logout_redirect_uri` is configured in your client registration in the `PostLogoutRedirectUris` section and not the `RedirectUris` section.
